### DISCUSSION TITLE
Dexcom G7 Connection problems after upgrading to Android 15

### DISCUSSION THREAD

**Discussion by miiihi** [NightscoutFoundation/xDrip#3733] [Bug reports]
Hi,

after upgrading to Android 15 I have an unstable connection to the G7. Looks like that the authentication/authorization step is somehow not properly completed, so that xDrip then cycles between normally getting data, being stuck on 'Authorizing 2', auto removing pairing and initiating paring again...
Already tried removing xDrip and clean installing it again, but again same behaviour

Phone: Pixel 8, Android 15
xDrip: 9f769df-2024.10.11

<img src="https://github.com/user-attachments/assets/60168bcf-8fed-4fff-971b-e6ec260cde37" height="800px" />
<img src="https://github.com/user-attachments/assets/3421e04b-a0d8-4ab0-be0a-a35c81be1080" height="800px" />
<img src="https://github.com/user-attachments/assets/012506e7-3488-4952-b71c-0ab6bfb72e2f" height="800px" />

I can provide further debug info if needed, but I guess I'll need to downgrade back to Android 14 soon, as the current state is really not good.

**Comment by Navid200** [comment]
Intermittent connectivity with a Pixel phone almost always is because battery optimization is not disabled correctly.

And it needs to be done for both xDrip and Bluetooth.
https://navid200.github.io/xDrip/docs/BatteryOpt.html
https://navid200.github.io/xDrip/docs/BluetoothBatteryOpt.html

If you think that you have done it already, please post a screenshot of where you think the setting is.

**Comment by miiihi** [comment]
Thanks for the answer, but the battery optimization is disabled. See the screenshots bellow.

Some additional observations:
- the problems began with the first sensor activated on Android 15. Interestingly, with the sensor that was already paired when upgrading (ie, paired with Android 14) there were no connection problems
- usually the connection recovers after one or two missing readings, but not always. Over todays night, I stopped receiving data at about 4am, then when I woke up at about 9, I restarted the xDrip and it again immidiately recovered... (but not for long)
-  there are numerous "Error count reached: X" and red "BleDisconnectedException...ErrorMissingConsumer:accept" errors in the log. Is this last one not significant?


![Screenshot_20241026-101936](https://github.com/user-attachments/assets/a39f830e-519e-425e-99e1-51cff3988373)
![Screenshot_20241026-101959](https://github.com/user-attachments/assets/7d69c657-c92f-43aa-aef9-623f219a6866)
![Screenshot_20241026-102100](https://github.com/user-attachments/assets/43ed1470-7807-462d-902a-1b6bd7929ea5)
![Screenshot_20241026-102135](https://github.com/user-attachments/assets/c132f3e0-9ad6-455a-8314-600bff29b28c)
![Screenshot_20241026-102034](https://github.com/user-attachments/assets/945f06de-38bd-4f82-aa46-59dfa286224d)

**Comment by jamorham** [comment]
Can you enable ob1 debugging and submit logs with a suitable identifier if the problem persists:
https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector

**Reply by ninelore** [reply]
Sent logs via xdrips upload feature

**Comment by ninelore** [comment]
Hi everyone, [Dexcom acknonledged the Problem even with the stock app](https://www.dexcom.com/en-US/faqs/will-android-15-affect-my-dexcom-g7-cgm-system)

While a supposed dexcom dev on reddit claims its a bug in Androids bluetooth stack, a renowned custom ROM developer claimed otherwise, suggesting a bad implementation that got broken by security fixes and improvements made in A15

**Reply by nickb24** [reply]
Can you post the link to this discussion?

**Comment by Navid200** [comment]
@miiihi Please follow JamOrHam's instructions.

Just to have covered everything, would you please also post a screenshot of the following page?
Settings > Dex Debug Settings

**Reply by ninelore** [reply]
Same issue, might as well post aswell
[Screenshot_20241026-131244.png](https://github.com/user-attachments/assets/03f9740f-16f7-4321-9dca-327c69c12060)

**Reply by Navid200** [reply]
@ninelore Thanks for posting the link to the Dexcom article.
Please don't post screenshots in this thread.
It is hard work to keep xDrip going.  Please help us.  By posting screenshots here, you are only making it harder for us to focus on one device.

You are more than welcome to open another thread.

**Reply by ninelore** [reply]
sry about that, turned screenshot into a link

i do believe its the same issue

**Reply by Navid200** [reply]
If you open another thread, and post in it some identifying factor from your upload, JamOrHam can interact with you and ask you for more info.
As he goes back and forth between your posts and his tests, it will be easy for him to find what he needs.
Please include in the first post in your thread the phone brand and model as well.

**Comment by miiihi** [comment]
@Navid200 @jamorham 
Here the Settings > Dex Debug Settings [screenshot-094723](https://github.com/user-attachments/assets/16461f6d-dfa3-443c-ac4e-8b3a256f9b2f)

Yesterday, the connection somewhat "stabilised". I didn't notice any missed readings during the day, although still a lot of errors and retries in the log, but looks like xDrip managed to get the data every 5 mins anyway.

I noticed the problems in the evening, didn't get any readings for an hour... Uploaded the logs (miiihi). Hard restarting the app helped to get the connection/data back immidiately! Uploaded the logs again after the restart...

**Comment by miiihi** [comment]
Any updates here? Can you confirm the problem with Android 15? or is it just me, having problems...

**Reply by Navid200** [reply]
I am using Android 15 and have no problems.  However, I use Pixel 6a.
I suspect there may be something on Pixel 8 causing this.  But, I cannot be sure.
Please be patient.  We have to wait for JamOrHam to get back to us as this could be complicated.

**Reply by databoy2k** [reply]
@miiihi, and at the risk of jumping on the "Me Too" train, I am also having the same issues. Same errors as well. Pixel 8, Android 15. I've had the problem with two separate sensors. The fix, irritatingly, is to clear settings, delete all former pairings, and re-initialize xDrip.

**Comment by Navid200** [comment]
The reason Pixel has been so good (compared to Samsung) is that Android on Pixel is not modified to effectively something different.
If there is something special about Pixel 8, we may need to do something to xDrip.  That will be a bad day.  Any day we have to change xDrip to address only one phone brand, we have taken a step backwards.  So, I hope that way will never come.

In the meantime, there is always a possibility that some app on your phone is the culprit.
Is any of you willing to restore the phone back to factory settings?
Then, only update Android.
Then, only install xDrip and set up existing apps like gmail or Chrome etc.  But, do NOT install any app other than xDrip.
What happens?  Do you experience the same problems?

If no, it proves the problem is caused by an app.
If yes, it proves the issue is fundamental with Pixel 8.

**Reply by ninelore** [reply]
Keep in mind that Pixels are the only phones with Android 15 atm. Theres a chance that this issue will appear on other brands once those catch up

**Reply by Navid200** [reply]
What I suggested helps us confirm something instead of speculate and talk about "there is a chance ...".

**Reply by ninelore** [reply]
It isnt pure speculation is that device manufacturer are probably going to include the problematic AOSP change which I mentioned somewhere already.

I've tried these steps already on 2 Phones (6 Pro and 9 Pro), both Stock and custom ROM (Graphene), Xdrip and Dexcom App: same result

**Reply by ninelore** [reply]
(Warning: actual speculation this time)

Based on the wording I recieved from Android OS developers I also wouldnt rule out that Dexcom might need to fix Sensor firmware

**Reply by Navid200** [reply]
I am using xDrip on Pixel 6a as a collector.  There is no problem at all.  
There wasn't with 15 beta.
There isn't any problem with 15.

**Reply by ninelore** [reply]
So far i heard of 2 no-problem cases (yours included) against dozens of Reddit posts and comments. Please don't rule out stuff just because of your personal experience.

**Reply by Navid200** [reply]
I don't rule it out.
I am trying to figure out what's going on. 

These are the instructions for using xDrip as a collector with G7:
https://navid200.github.io/xDrip/docs/Dexcom/G7.html

It is possible that an app is causing the problems.  I suggested what to do to rule it out.

We have always improved the collector and we will continue to do so.

I don't have anything else to say.  I have tried to focus on the subject and not make it personal.
I advise everyone else to do the same.

**Reply by databoy2k** [reply]
Unfortunately, I can't reset at this time. Too many security apps that are tying themselves to the current instance. On my next sensor I could potentially go back to a Pixel 6 and carry two phones for 10 days. That's not for another week though.

Is it possible to implement the error handler described in the error? It might give us a better idea what the problem is.

**Reply by Navid200** [reply]
@databoy2k It is not practical to have a conversation (not a broadcast) with multiple people who believe are experiencing the exact same issue attempting to troubleshoot something remotely.

I have just updated the guide today adding more detail.  
The best thing you can do is to follow the guide, update to the latest Nightly, enable OB1 debugging log details, start a new thread including phone brand and model in the very first post, upload logs, and post in the thread including an identifying phrase you used when uploading the logs.

Then, when you post in that thread stating `I can't reset`, it will be clear what you are talking about just as an example.  
Also when you see a post in that thread, you can be sure we are talking to you whereas you have now responded to a post that was not addressing you at all.

**Reply by databoy2k** [reply]
> Then, when you post in that thread stating `I can't reset`, it will be clear what you are talking about just as an example. Also when you see a post in that thread, you can be sure we are talking to you whereas you have now responded to a post that was not addressing you at all.

Navid, your work on this project (and JamOrHam's) is invaluable to most of us living with this GD disorder. I appreciate everything that you're doing.

I didn't reply to your argument; I replied to your post on the discussion. Assuming that everyone posting (and the lurkers) are all together trying to resolve what isn't just a minor toy but rather health-critical software, I wanted to be clear that I couldn't answer your challenge here (https://github.com/NightscoutFoundation/xDrip/discussions/3733#discussioncomment-11101775) and instead that someone else could hopefully give that a shot. 

This is kind of the nature of having these chats in "Discussions" rather than in "Issues" - you end up with several threads like this one. 

All that aside, I'll take a look at the updated guide. I've been uploading logs, as you've no doubt seen. There's no need to make you guys try to keep track of several identical "discussions" of the exact same issue. 

Hopefully it'll get sorted. As I indicated earlier, I'm able to sort it out on sensor changes, even if it takes me literal days to get my settings back the way I like them (haha).

**Comment by Pentagit** [comment]
I'm on a Pixel 8 Pro with Android 15 and also have had this connection problems a few days ago with G7.
Only solution was to install the Dexcom app from the store and let it search for the sensor, which took about 3 hours (!) but it finally connected and I didn't have any hiccups so far. But the next new sensor is only a few days around the corner... so it would be nice to know that there will be a solution soon. At least also Dexcom seems to have taken notice by this problem (as for your link above @ninelore ). If I can provide any information for you let me know!

**Comment by nsghub** [comment]
I can confirm this issue on my Pixel 7a.

**Comment by Navid200** [comment]
There is a new update for Android 15.
I have just updated my Pixel 6a to it.  It is still working fine with G7 with no connectivity issues.

**Reply by Pentagit** [reply]
I have updated as well.
https://support.google.com/pixelphone/thread/305744117/google-pixel-update-november-2024

Google states that this update contains a 
"Fix for issue with Bluetooth range under certain conditions*[3]"

But only for Pixel 9 phones. We will have to see if this fix applies to Pixel 8 phones as well.

**Reply by Pentagit** [reply]
Update: Last sensor after android patch connected instantly with dexcom app. Didn't test it with xdrip though.

**Comment by Unrealsweb** [comment]
Are there any new findings? My Sony Xperia 1 VI is now also getting the Android 15 update.

**Reply by Pentagit** [reply]
Haven't had any more connection problems with original dexcom app and xdrip companion.

**Comment by Navid200** [comment]
Is anyone here using G6 and experiencing problems with it on Android 15?

Or are problems related to G7 only?
Have you seen this?
https://github.com/NightscoutFoundation/xDrip/discussions/3813

**Comment by miiihi** [comment]
Sorry, on G7
But forgot to update this discussion. After first two sensors, the problems simply disapeared. Working flawlessly ever since. Possibly related to the November Pixel Android Update, although changelog doesn't show anything relevant.

**Comment by Unrealsweb** [comment]
I have the same problem now on my Sony Xperia 1 VI after updating to Android 16 and tried everything here without success :-(

**Reply by Navid200** [reply]
Android 16 has been out for months.  We tested it when it came out year ago in beta.  There are so many people using it with G7 already.
These are the troubleshooting steps: https://navid200.github.io/xDrip/docs/Dexcom/G7_Troubleshooting.html

**Reply by Navid200** [reply]
If you follow the guidelines and it does not work, open a new discussion and I will see what could be causing the problem.
